samples/drivers/display/boards/da1469x_dk_pro.overlay

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <zephyr/dt-bindings/display/ili9xxx.h>
+ #include <zephyr/dt-bindings/gpio/gpio.h>
+
+ / {
+    chosen {
+        zephyr,display = &ili9340;
+    };
+ };
+
+ &mipi_dbi {
```

Comment:
Is the display used here an external module, or integrated with the board? If integrated, it might make sense to define it in the board devicetree instead of as an overlay

Reply:
I was sort of unsure how to handle this as the specific display module is neither integrated with the board nor promoted for the specific display interface. Therefore, I decided to move its configuration in overlay files. However, I will need to safeguard it against other display interfaces/modules as the same LCD controller might also be used to drive [display panels](https://github.com/zephyrproject-rtos/zephyr/pull/67649/commits/31dfb94fc980cbce0db9354f6c92e8f5054ba64c). To be honest, I have not encountered related use cases so, I guess it would be ok to introduce a Kconfig choice in our board's Kconfig.defconfig which should select the display node/module. Let me know if there are other approaches for this to work.

Reply:
Is this a case where the module could be a [shield](https://docs.zephyrproject.org/latest/hardware/porting/shields.html)? Generally anything using the Arduino interface is a great candidate for this. If the interface is completely nonstandard, keeping the display as an overlay is probably best. NXP has used shields to define displays when there is a shared interface used between multiple display types (we use a 40 pin FPC for several modules) so this could be another option.

Reply:
To be clear here- the LCD controller pins are exposed on the arduino header, in the same location as the SPI pins are usually exposed? I'm not certain what the best way to handle this is, because the `adafruit_2_8_tft_touch_v2` shield assumes a standard SPI peripheral is exposed via the SPI pins on the arduino header (so it will use the MIPI DBI SPI driver). This might be a case where using a board overlay is the best we can do.

Reply:
Yes, the LCD controller's pins are exposed in the same location as the SPI pins. So, my understanding is that we should stick with my initial approach, correct? In that case, I would like to ask what is the best way to add support for multiple LCD interfaces within the same `<BOARD>.overlay` file. A tried to use Kconfig symbols along with pre-processor directives but it seems that Kconfigs are not interpreted correctly (not sure if it's my fault or it's that such a scheme is not allowed in overlays.)